openmdao/utils/units.py

```diff
@@ -1052,6 +1054,31 @@ def convert_units(val, old_units, new_units=None):
     return (val + offset) * factor
 
 
+def _has_val_mismatch(units1, val1, units2, val2):
+    """
+    Return True if values differ after unit conversion or if values differ when units are None.
+
+    Parameters
+    ----------
+    units1 : str or None
+        Units for first value.
+    val1 : float or ndarray
+        First value.
+    units2 : str or None
+        Units for second value.
+    val2 : float or ndarray
+        Second value.
+    """
+    if units1 != units2:
+        if units1 is None or units2 is None:
+            return True
+
+        # convert units
+        val1 = convert_units(val1, units1, new_units=units2)
+
+    return np.linalg.norm(val1 - val2) > 1e-10
```

Comment:
Wonder if this comparison should be a relative difference instead of absolute -- for cases where we have 2 very large numbers (like maybe modulus of elasticity.)

Reply:
done